### PR TITLE
[Java] Iterating Over Multiple Subscriptions

### PR DESCRIPTION
**Pull request by MauriceVanVeen**
Resolves https://github.com/ConnectEverything/nats-by-example/issues/245

### CODE REVIEW

File: examples/messaging/iterating-multiple-subscriptions/java/Main.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+public class Main {
+    public static void main(String[] args) {
+        String natsURL = System.getenv("NATS_URL");
+        if (natsURL == null) {
+            natsURL = "nats://127.0.0.1:4222";
+        }
+
+        // Initialize a connection to the server. The connection is AutoCloseable
+        // on exit.
```
Comment (scottf) [inline examples/messaging/iterating-multiple-subscriptions/java/Main.java:18]: You can remove this, it's not in any other example.

File: examples/messaging/iterating-multiple-subscriptions/java/Main.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        try (Connection nc = Nats.connect(natsURL)) {
+
+            // Create a message dispatcher for handling messages in a
+            // separate thread and then subscribe to multiple target subjects.
```
Comment (scottf) [inline examples/messaging/iterating-multiple-subscriptions/java/Main.java:22]: Would you replace the comment with this or something like this:
```
// Create a message dispatcher. A dispatcher a process that runs on
// its own thread, receives incoming messages via a FIFO queue,
// for subjects registered on it. For each message it takes from
// the queue, it makes a blocking call to the MessageHandler
// passed to the createDispatcher call.
```

File: examples/messaging/iterating-multiple-subscriptions/java/Main.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            Dispatcher dispatcher = nc.createDispatcher((msg) -> {
+                System.out.printf("Received %s: %s\n",
+                        msg.getSubject(),
+                        new String(msg.getData(), StandardCharsets.UTF_8));
+            });
+
```
Comment (scottf) [inline examples/messaging/iterating-multiple-subscriptions/java/Main.java:28]: maybe add a comment before the subscribes
```
// subscribe directly on the dispatcher by subject
```

File: examples/messaging/iterating-multiple-subscriptions/java/Main.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                nc.publish("s1", String.valueOf(i).getBytes(StandardCharsets.UTF_8));
+                nc.publish("s2", String.valueOf(i).getBytes(StandardCharsets.UTF_8));
+                nc.publish("s3", String.valueOf(i).getBytes(StandardCharsets.UTF_8));
+                nc.publish("s4", String.valueOf(i).getBytes(StandardCharsets.UTF_8));
+                Thread.sleep(100);
+            }
+
+            // Sleep this thread a little so the dispatcher thread has time
+            // to receive all the messages before the program quits.
+            Thread.sleep(200);
```
Comment (scottf) [inline examples/messaging/iterating-multiple-subscriptions/java/Main.java:46]: Can you bump this to 500? Alternatively you could use a CountDownLatch in the handler and wait for it to complete.

**Review by scottf — CHANGES_REQUESTED** [review]

File: examples/messaging/iterating-multiple-subscriptions/java/Main.java (hunk 1 above)
Comment (MauriceVanVeen) [reply examples/messaging/iterating-multiple-subscriptions/java/Main.java:18]: When looking at the other available examples, Rust & .NET v2, it's indeed not there.
But for the Java examples for both pub/sub and request/reply it is there.

Would you like me to remove this for these new Java examples, or keep it in for consistency in the Java examples?

File: examples/messaging/iterating-multiple-subscriptions/java/Main.java (hunk 2 above)
Comment (MauriceVanVeen) [reply examples/messaging/iterating-multiple-subscriptions/java/Main.java:22]: Done

File: examples/messaging/iterating-multiple-subscriptions/java/Main.java (hunk 3 above)
Comment (MauriceVanVeen) [reply examples/messaging/iterating-multiple-subscriptions/java/Main.java:28]: Added it, rephrased a little bit:
```java
// Subscribe directly on the dispatcher for multiple subjects.
```

File: examples/messaging/iterating-multiple-subscriptions/java/Main.java (hunk 4 above)
Comment (MauriceVanVeen) [reply examples/messaging/iterating-multiple-subscriptions/java/Main.java:46]: Done, using a `CountDownLatch` instead.

**Comment by MauriceVanVeen** [comment]
@scottf, have applied your feedback in this PR which does build/deploy correctly: https://github.com/ConnectEverything/nats-by-example/pull/253

Closing this one.

**Pull request closed without merging**